exonum/src/runtime/dispatcher/mod.rs

```diff
@@ -415,25 +439,78 @@ impl Dispatcher {
         self.register_running_service(&instance);
         Ok(())
     }
+
+    /// Perform a configuration update with the specified changes.
+    fn update_config(
+        &self,
+        fork: &mut Fork,
+        caller_instance_id: InstanceId,
+        changes: Vec<ConfigChange>,
+    ) {
+        // An error while configuring one of the service instances should not affect others.
+        changes.into_iter().for_each(|change| match change {
+            ConfigChange::Consensus(config) => {
+                trace!("Updating consensus configuration {:?}", config);
+
+                blockchain::Schema::new(fork as &Fork)
+                    .consensus_config_entry()
+                    .set(config);
+                fork.flush();
+            }
+
+            ConfigChange::Service(config) => {
+                trace!(
+                    "Updating service instance configuration, instance ID is {}",
+                    config.instance_id
+                );
+
+                let configure_result = catch_panic(|| {
+                    let dispatcher_ref = DispatcherRef::new(self);
+                    let context = CallContext::new(
+                        fork,
+                        &dispatcher_ref,
+                        caller_instance_id,
+                        config.instance_id,
+                    );
+                    ConfigureCall::from(context).apply_config(config.params)
+                });
+
+                match configure_result {
```

Comment:
Why the call of `Configure::apply_config` method is different from any other service method? Here you have an entirely duplicated piece of code from `execute`, don't you?

Reply:
Yes, but execute method is only for transactions and, secondly, behavior on errors is different.